tests/test_repository.py

```diff
@@ -3,315 +3,353 @@
+    def test_undefined_repo_id(self, repo):
+        """Failure case: access undefined repository ID."""
+        repo.repo_id = None
+        with pytest.raises(
+            RuntimeError, match="Attempting to access undefined repo ID"
+        ):
+            _ = repo.repo_id
 
 
-def test_checkout():
+class TestCheckout:
     """Tests for `CableRepository.checkout()`."""
-    # Success case: checkout mock repository
-    mock_subprocess = MockSubprocessWrapper()
-    repo = get_mock_repo(mock_subprocess)
-    repo.checkout()
-    assert (
-        f"svn checkout https://trac.nci.org.au/svn/cable/trunk {MOCK_CWD}/src/trunk"
-        in mock_subprocess.commands
-    )
 
-    # Success case: checkout mock repository with specified revision number
-    mock_subprocess = MockSubprocessWrapper()
-    repo = get_mock_repo(mock_subprocess)
-    repo.revision = 9000
-    repo.checkout()
-    assert (
-        f"svn checkout -r 9000 https://trac.nci.org.au/svn/cable/trunk {MOCK_CWD}/src/trunk"
-        in mock_subprocess.commands
-    )
+    def test_checkout_command_execution(self, repo, mock_cwd, mock_subprocess_handler):
+        """Success case: `svn checkout` command is executed."""
+        repo.checkout()
+        assert (
+            f"svn checkout https://trac.nci.org.au/svn/cable/trunk {mock_cwd}/src/trunk"
+            in mock_subprocess_handler.commands
+        )
 
-    # Success case: test non-verbose standard output
-    mock_subprocess = MockSubprocessWrapper()
-    repo = get_mock_repo(mock_subprocess)
-    with contextlib.redirect_stdout(io.StringIO()) as buf:
+    def test_checkout_command_execution_with_revision_number(
+        self, repo, mock_cwd, mock_subprocess_handler
+    ):
+        """Success case: `svn checkout` command is executed with specified revision number."""
+        repo.revision = 9000
         repo.checkout()
-    assert (
-        buf.getvalue()
-        == f"Successfully checked out trunk at revision {mock_subprocess.stdout}\n"
-    )
+        assert (
+            f"svn checkout -r 9000 https://trac.nci.org.au/svn/cable/trunk {mock_cwd}/src/trunk"
+            in mock_subprocess_handler.commands
+        )
 
-    # Success case: test verbose standard output
-    mock_subprocess = MockSubprocessWrapper()
-    repo = get_mock_repo(mock_subprocess)
-    with contextlib.redirect_stdout(io.StringIO()) as buf:
-        repo.checkout(verbose=True)
-    assert (
-        buf.getvalue()
-        == f"Successfully checked out trunk at revision {mock_subprocess.stdout}\n"
-    )
+    def test_standard_output(self, repo, mock_subprocess_handler):
+        """Success case: test standard output."""
+        with contextlib.redirect_stdout(io.StringIO()) as buf:
+            repo.checkout()
+        assert (
+            buf.getvalue()
+            == f"Successfully checked out trunk at revision {mock_subprocess_handler.stdout}\n"
+        )
+        with contextlib.redirect_stdout(io.StringIO()) as buf:
```

Comment:
```suggestion
       # Verbose output
        with contextlib.redirect_stdout(io.StringIO()) as buf:
```
It's hard to see why the test is repeated otherwise.

Or parameterise on verbosity...

Reply:
Good spotting, looks like I forgot to parametrise this one...